src/Middleware/HttpLogging/src/HttpLoggingContext.cs

```diff
@@ -0,0 +1,93 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.AspNetCore.Http;
+
+namespace Microsoft.AspNetCore.HttpLogging;
+
+/// <summary>
+/// The context used for logging customization callbacks.
+/// </summary>
+/// <param name="httpContext">The request context.</param>
+public sealed class HttpLoggingContext(HttpContext httpContext)
+{
+    /// <summary>
+    /// The request context.
+    /// </summary>
+    public HttpContext HttpContext { get; internal set; } = httpContext;
+
+    /// <summary>
+    /// What parts of the request and response to log.
+    /// </summary>
+    public HttpLoggingFields LoggingFields { get; set; }
+
+    /// <summary>
+    /// Limits how much of the request body to log.
+    /// </summary>
+    public int RequestBodyLogLimit { get; set; }
+
+    /// <summary>
+    /// Limits how much of the response body to log.
+    /// </summary>
+    public int ResponseBodyLogLimit { get; set; }
```

Comment:
Is there a requirement for these to be changeable here? The problem with setting these values is they don't compose. It would be interceptors potentially overwriting each other's values. Why would someone want to set them here vs the current configuration?

Reply:
The advantage of the callback is being able to make more dynamic/granular choices. E.g. give admins a higher limit. That's not the kind of thing that can be expressed in the existing config.

I'm less worried about conflicts since these are expected to be provided by the application. If they have conflicts in their business logic that's up to them to sort out.